### DISCUSSION TITLE
UUIDs don't generate randomly

### DISCUSSION THREAD

**Discussion by AlejandroSuero** [planetscale/discussion#542] [Q&A] [answered]
I am doing a REST API that retrieves movies and they are identified by an id wich is an UUID.

I am using javascript and MySQL and when I deployed the database to PlanetScale, my code for generating UUIDs when creating a POST was not working.

When I used that sentence in PlanetScale's cli, I got this:

```bash
dbname/|? main ?|> select UUID() uuid;
+--------------------------------------+
| uuid                                 |
+--------------------------------------+
| 88e5455b-56ec-11ee-8168-6e9e1be6f14c |
+--------------------------------------+
dbname/|? main ?|> select UUID() uuid;
+--------------------------------------+
| uuid                                 |
+--------------------------------------+
| 88e5455b-56ec-11ee-8168-6e9e1be6f14c |
+--------------------------------------+
dbname/|? main ?|> select UUID() uuid;
+--------------------------------------+
| uuid                                 |
+--------------------------------------+
| 88e5455b-56ec-11ee-8168-6e9e1be6f14c |
+--------------------------------------+
dbname/|? main ?|> select UUID() uuid;
+--------------------------------------+
| uuid                                 |
+--------------------------------------+
| 88e5455b-56ec-11ee-8168-6e9e1be6f14c |
+--------------------------------------+
dbname/|? main ?|>
```

As you can see, it keeps generating the same UUID instead of a new one. Is there a solution for this case? I didn't see anything on the docs related to this, so I'd apreciate any help.

> NOTE: I am not using any ORM so every query is written in MySQL, my MySQL model uses the local config if PlanetScale is not available or I am doing some tests.

```javascript
const [uuidResult] = await connection.query("SELECT UUID() uuid;")
let [{ uuid }] = uuidResult
const [uuidFromDB] = await connection.query(
    "SELECT BIN_TO_UUID(id) id FROM movie WHERE id = UUID_TO_BIN(?)",
    uuid
)
if (uuidFromDB.length > 0) {
    if (uuid === uuidFromDB[0].id) uuid = crypto.randomUUID()
}
```

> NOTE: I came up with this way using Node's uuid because of the issue

**Comment by orware** [comment]
Hello @AlejandroSuero,

I've ran some tests and observed the same behavior you did when running the query via our web console or via `pscale shell`.

To confirm, with your node.js code the same issue with the query was being observed with the same value being returned each time?

If so, I think that needs to be investigated a little further on our end so I'll be filing an issue report about that internally.

When I ran the `SELECT UUID() uuid;` query from a MySQL GUI client, DBeaver in this case, the results were always unique on every execution of the query so there may be something unique about connections coming from the other two options that will explain the difference in behavior once the team has a chance to investigate a little further.

On your end, it looks like you are using the node.js `mysql2` library to connect to your database, but if you can confirm that is accurate that would be helpful.

I'll keep you posted on any updates I receive from our end!

**Comment by AlejandroSuero** [comment]
Thanks for the quick response @orware ,

Yes, I am indeed using ```mysql2``` with nodejs.

```bash
$ node --version
v18.17.1
$ npm --version
10.1.0
```

I notice the error first when using cURL command to hit my API. And then checking the query it was the same uuid everytime.

I used, DBeaver, MySQL Workbench and planetscale-cli. Same result, same uuid.

If I can help in any way, feel free to reach out.

**Reply by orware** [reply, marked as answer]
Hello @AlejandroSuero,

I reported the issue yesterday afternoon and our Vitess team has already implemented a fix as a part of this pull request within the public Vitess project:
https://github.com/vitessio/vitess/pull/14051

There will likely be a delay before the fix is officially included in the Vitess images we deploy within PlanetScale so I don't believe it will be immediately available but within a few weeks it will hopefully be included on our end, allowing you to use the `SELECT UUID() uuid;` query you shared without modifications.

In the meanwhile, I was just trying out a potential workaround, which uses one of the [Vitess comment directives](https://vitess.io/docs/17.0/user-guides/configuration-advanced/comment-directives/#skip-query-plan-cache-skip_query_plan_cache), and seems to allow things to behave as expected more immediately which you may be able to give a try for now:

```
SELECT /*vt+ SKIP_QUERY_PLAN_CACHE */ UUID() uuid;
```

**Reply by AlejandroSuero** [reply]
Hello @orware,

Thank you for assisting me in this issue, I tried the query you sent me and is working in my end as well.

Hope you have a wonderfull day and keep up the work you do 😄.
